Realm.Shared/RealmList.cs

```diff
@@ -183,7 +183,7 @@ public int Count
         public void Add(T item)
         {
             this.ManageObjectIfNeeded(item);
-            var rowIndex = item.RowHandle.RowIndex;
+            var rowIndex = item.ObjectHandle.RowIndex;
             _listHandle.Add(rowIndex);
```

Comment:
Consider passing the `ObjectHandle` directly instead of fetching the row index in C# to save one extra native call. You can/should do that in a follow-up PR, though.